docs/sdk/src/reference_docs/frame_currency.rs

```diff
@@ -1,8 +1,51 @@
-//! FRAME Currency Abstractions and Traits
+//! # FRAME Currency Abstractions and Traits
 //!
-//! Notes:
+//! Currency related traits in FRAME provide standardized interfaces for implementing various
+//! currency functionalities. These traits enable developers to create, transfer, and manage
+//! different forms of digital assets within the blockchain environment, ensuring that the economic
+//! aspects of the chain are robust and adaptable to different use cases.
 //!
-//! - History, `Currency` trait.
-//! - `Hold` and `Freeze` with diagram.
-//! - `HoldReason` and `FreezeReason`
-//! 	- This footgun: <https://github.com/paritytech/polkadot-sdk/pull/1900#discussion_r1363783609>
+//! ## The Currency Trait
+//!
+//! The [`Currency`](../../../frame_support/traits/tokens/currency/index.html) trait was initially
+//! introduced in Substrate to manage the native token balances. This trait was later deprecated in
+//! favor of the [`fungible`](../../../frame_support/traits/tokens/fungible/index.html) traits in
+//! Substrate's PR [#12951](https://github.com/paritytech/substrate/pull/12951). This shift is part
+//! of a broader initiative to enhance token management capabilities within the framework. This
+//! deprecation is aimed at providing improved safety and more flexibility for managing assets,
+//! beyond the capabilities of the original
+//! [`Currency`](../../../frame_support/traits/tokens/currency/index.html) trait. This transition
+//! enables more diverse economic models in Substrate. For more details, you can view the discussion
+//! on the [Tokens Horizon issue](https://github.com/paritytech/polkadot-sdk/issues/327). The
+//! [`Currency`](../../../frame_support/traits/tokens/currency/index.html) trait is still available
+//! in Substrate, but it is recommended to use the **fungible** traits instead. The
+//! [deprecation PR](https://github.com/paritytech/substrate/pull/12951) has a dedicated section on
+//! upgrading from **Currency** to **fungible**. Besides, this [issue](https://github.com/paritytech/polkadot-sdk/issues/226)
+//! lists the pallets that have been upgraded to the **fungible** traits, and the ones that are
+//! still using the [`Currency`](../../../frame_support/traits/tokens/currency/index.html) trait.
+//! There one could find the relevant code examples for upgrading.
+//!
+//! ## Fungible Traits
```

Comment:
You are not going to briefly talk about `Fungibles`?

Reply:
I see `fungibles` making sense in multiple assets scenario but not here, there is this other [PR](https://github.com/paritytech/polkadot-sdk/pull/2802) that covers more generic tokens topics